docs/azure.md

```diff
@@ -0,0 +1,116 @@
+
+## Create Azure VMs
+We first need to allocate the VMs. We provide a script
+```bash
+./create_vms.sh
+```
+to create VMs with the Azure SKU in the region specified in `azure_config.json`. Feel
+free to customize your JSON to your desired region/SKU. This step will take a few minutes
+to complete while it sets up all of your VMs on Azure.
+
+## Setup VM environment to use DeepSpeed
+Next, we need to configure the VM environment for DeepSpeed. We provide a script
+```bash
+./setup_vms.sh
+```
+to generate a [hostfile](../README.md#resource-configuration) and SSH
+configuration on all of the VMs. This configuration will be used by the DeepSpeed
+Docker containers in the next step.
+
+## Start the DeepSpeed docker container
+We now setup the DeepSpeed Docker containers on the VMs. We provide a script
+```bash
+./setup_docker.sh
+```
+to pull the DeepSpeed image onto all VMs and start a container instance in the
+background. This will take several minutes since it needs to pull the entire Docker
+image.
+
+## Access VMs
+The tool [azure_ssh.sh](azure_ssh.sh) will let you SSH into any of the VMs with this
+syntax:
+```bash
+./azure_ssh.sh <node-id> [command]
+```
+where the `node-id` is a number between `0` and `num_vms-1`.  This script will find the
+public IP address of your VM and use the SSH key provided in the Azure configuration
+JSON.
+
+## Access DeepSpeed container
+Everything should be up and running at this point. Set's access the running DeepSpeed
+container on the first VM and make sure we can talk to the other containers in our setup.
+Let's complete the following steps:
+
+ * SSH into the first VM via: `./azure_ssh.sh 0`
+ * Change directories into the azure folder of this repo via: `cd ~/workdir/DeepSpeed/azure`
+ * Attach the running docker container via: `./attach.sh`
+ * You should now be able to `ssh` into any other docker container, the containers can be
+   accessed via their SSH alias of `worker-N`, where `N` is the VM number between `0`
+   and `num_vms-1`. In this example we should be able to successfully run `ssh worker-1
+   hostname`. You can also use `ds_ssh` to execute a command in parallel on all of your
+   worker containers.
+
+## Run CIFAR-10 example model
+We will now run the DeepSpeed CIFAR-10 model example to test the VM setup. From inside
+the first DeepSpeed container:
+
+  1) Install the python dependencies necessary to run the CIFAR-10 example model. You can
+  do this across your cluster via:
+  ```bash
+  ds_ssh pip install -r ~/workdir/DeepSpeed/DeepSpeedExamples/cifar/requirements.txt
+  ```
+
+  2) Now change directories to the CIFAR example:
+  ```bash
+  cd ~/workdir/DeepSpeed/DeepSpeedExamples/cifar
+  ```
+
+  3) Finally, launch training across all VMs:
+  ```bash
+  deepspeed cifar10_deepspeed.py --deepspeed --deepspeed_config ds_config.json
+  ```
+  Alternatively, we provide a helper script `./run_ds.sh`.
+
+This will train a simple CIFAR-10 example model. The accuracy that you will achieve will
```

Comment:
Can we rephrase this comment? It requires a subtle understanding of data parallelism and I'm worried it will be misinterpreted as a critique of DeepSpeed.